File: app/src/main/java/com/github/batulovandrey/urbandictionarycom/adapter/AlphabetViewHolder.java
package com.github.batulovandrey.urbandictionarycom.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.github.batulovandrey.urbandictionarycom.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * @author Andrey Batulov on 27/10/2017
 */

public class AlphabetViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    @BindView(R.id.alphabet_text_view)
    TextView mAlphabetTextView;

    private AlphabetClickListener mClickListener;

    AlphabetViewHolder(View itemView, AlphabetClickListener listener) {
        super(itemView);
        mClickListener = listener;
        ButterKnife.bind(this, itemView);
        itemView.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        mClickListener.onItemClick(getLayoutPosition());
    }
}

File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/adapter/AlphabetViewHolder.kt
package com.github.batulovandrey.unofficialurbandictionary.adapter

import android.support.v7.widget.RecyclerView
import android.view.View
import kotlinx.android.synthetic.main.alphabet_item.view.*

/**
 * @author Andrey Batulov on 22/12/2017
 */

class AlphabetViewHolder internal constructor(itemView: View, private val mClickListener: AlphabetClickListener)
    : RecyclerView.ViewHolder(itemView), View.OnClickListener {

    init {
        itemView.setOnClickListener(this)
    }

    fun setAlphabetText(text: String) {
        itemView.alphabet_text_view.text = text
    }

    override fun onClick(view: View) {
        mClickListener.onItemClick(layoutPosition)
    }
}